### ISSUE TITLE
Git: After cloning and no folder is open, add option to open the folder automatically.

### ISSUE THREAD

**Issue by jtagcat** [microsoft/vscode#93300]
v1.43.1
![image](https://user-images.githubusercontent.com/38327267/77420124-693de080-6dc1-11ea-959b-7687e9171029.png)
Add an option in preferences, that when this message appears and you don't have anything open, open the folder automatically.

**Comment by joaomoreno** [comment]
I like it. Let's just open, instead of adding an option.

**Comment by jtagcat** [comment]
> I like it. Let's just open, instead of adding an option.

If something is already open, opening a folder in the same window may close other stuff, that's probably why the notification exists in the first place and why I would recommend a 'default' option.

**Comment by ruwaid4** [comment]
Hello, me and my friend are thinking about doing this issue for an assignment. Is there anything we should know in particular about this issue?

**Comment by jtagcat** [comment]
I would find any information on why the notification exists in the first place.

Because the cloned repo would've been opened by default, if there wasn't the 'open in new window option'.

There's also an option to move this question to the prompt after asking for what to clone and where to clone.
(the prompt being this thing)
![image](https://user-images.githubusercontent.com/38327267/78285005-c3e8e200-750f-11ea-8280-f0ecf654a3f9.png)

**Comment by eXotech-code** [comment]
Hi there. I wanted to help with this issue as my first contribution to this repository.
But I have been stuck on one thing for a long time now.
I created a setting in package.json called "git.promptToOpenClonedRepo" of a type "boolean" and set it's default value to: "true".
The problem is that I do not see this setting when I open defaultSettings.json file after running my build and after cloning and opening a repository, I get this message:
![error message](https://user-images.githubusercontent.com/58250697/78343566-4ff41b80-759b-11ea-8edb-f75c02edf78a.png)
Am I doing something wrong? Is there a different file than package.json that stores git extension settings?

**Comment by eXotech-code** [comment]
Hello, I wanted to try my hand on this issue as my first contribution to this project.
So I have made a new setting named "Prompt to Open Cloned Repository" of type string and set the default value to 'showPrompt'. I have added four different settings: openNewWindow - always opens cloned repo in new window, openCurrentWindow - same but for current window, noFolderOpened - opens in current window only if there is no folder opened in current workspace, showPrompt - always shows prompt (default behaviour).
![setting](https://user-images.githubusercontent.com/58250697/78356796-88eabb00-75b0-11ea-9cb7-80c6892181a2.jpg)
I will add a pull request soon.
